data/mods/megamons/scripts.js

```diff
@@ -17,10 +17,10 @@ let BattleScripts = {
 				case 'Mega':
 				case 'Mega-X':
 				case 'Mega-Y':
-					this.modData('Pokedex', i).gen = 7;
-					this.modData('Pokedex', i).isMega = true;
-					this.modData('Pokedex', i).battleOnly = false;
-					delete this.modData('FormatsData', i).requiredItem;
+					if (this.modData('Pokedex', i).species === 'Crucibelle-Mega') break;
```

Comment:
Add 'Crucibelle-Mega' to the list above instead?

Also, I think the problem the validation code was trying to fix is that setting `battleOnly` to false here doesn't actually work because dex-data sets it back to true when it sees that `isMega` is true.

Reply:
Is there a reason to set IsMega to true?

Reply:
The reason why the code wasn't working before is because ``battleOnly`` is a ``FormatsData`` element and the modData was trying to edit it from the ``Pokedex`` element.

I don't think there's an actual reason to modify ``isMega`` at all but it doesn't harm the code to have it there.